tools/build_cpp.py

```diff
@@ -62,8 +73,12 @@ def main(*,
         '-DWITH_CTEST': 'OFF',
         '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION': ipo,
         '-DFULL_BUILD': 1,
+        '-DCMAKE_BUILD_TYPE': 'Debug' if debug_build else "Release",
     }
 
+    if debug_build:
+        cmake_flags["-DDYNAMIC_LIB"] = "ON"
```

Comment:
Do we have to enforce this or can MSVC also make a debug build with static libraries?